test/e2e/upgrades/apps/deployments.go

```diff
@@ -140,47 +134,9 @@ func (t *DeploymentUpgradeTest) Test(f *framework.Framework, done <-chan struct{
 
 	By(fmt.Sprintf("Waiting for deployment %q to complete adoption", deployment.Name))
 	framework.ExpectNoError(framework.WaitForDeploymentStatus(c, deployment))
-
-	By(fmt.Sprintf("Checking that replica sets for deployment %q are the same as prior to the upgrade", t.updatedD.Name))
```

Comment:
Checking whether the replicaset is the same as prior to upgrade has been performed by replicaset PR: #52449